wallet/ledger/ledger.go

```diff
@@ -87,14 +86,33 @@ func (af *ledgerAccountFactory) SupportedImportKinds() []wallet.ImportKind {
 }
 
 func (af *ledgerAccountFactory) HasConsensusSigner(rawCfg map[string]interface{}) bool {
-	return true
+	cfg, err := af.unmarshalConfig(rawCfg)
+	if err != nil {
+		return false
+	}
+
+	switch cfg.Algorithm {
+	case wallet.AlgorithmEd25519Legacy, wallet.AlgorithmEd25519Adr8:
+		return true
+	}
+	return false
 }
 
 func (af *ledgerAccountFactory) unmarshalConfig(raw map[string]interface{}) (*accountConfig, error) {
 	if raw == nil {
 		return nil, fmt.Errorf("missing configuration")
 	}
 
+	// CONFIG MIGRATION 1 (add legacy derivation support): Set default derivation to ADR 8, if not set.
```

Comment:
Will this ever be saved back? It would be good if the first time migration is performed the underlying config is actually updated.